contracts/tokens/SecurityToken.sol

```diff
@@ -394,6 +411,143 @@ contract SecurityToken is ERC20, ERC20Detailed, Ownable, ReentrancyGuard, Securi
+        // NB - passing `_additionalBalance` value is 0 because accessing the balance before transfer
+        uint256 balanceBeforeTransferLocked = _balanceOfByPartition(LOCKED, _to, 0);
+        _transferWithData(_from, _to, _value, _data);
+        // NB - passing `_additonalBalance` valie is 0 because balance of `_to` was updated in the transfer call
+        uint256 balanceAfterTransferLocked = _balanceOfByPartition(LOCKED, _to, 0);
+        toPartition =  _returnPartition(balanceBeforeTransferLocked, balanceAfterTransferLocked, _value);
+        emit TransferByPartition(toPartition, _operator, _from, _to, _value, _data, _operatorData);
+    }
+
+    function _returnPartition(uint256 _beforeBalance, uint256 _afterBalance, uint256 _value) internal pure returns(bytes32 toPartition) {
+        // return LOCKED only when the transaction `_value` should be equal to the change in the LOCKED partition
+        // balance otherwise return UNLOCKED
+        if (_afterBalance.sub(_beforeBalance) == _value)
+            toPartition = LOCKED;
+        // Returning the same partition UNLOCKED 
+        toPartition = UNLOCKED;
+    }
+
+    ///////////////////////
+    /// Operator Management
+    ///////////////////////
+    
+    /**
+     * @notice Authorises an operator for all partitions of `msg.sender`.
+     * NB - Allowing investors to authorize an investor to be an operator of all partitions
+     * but it doesn't mean we operator is allowed to transfer the LOCKED partition values.
+     * Logic for this restriction is written in `operatorTransferByPartition()` function.
+     * @param _operator An address which is being authorised.
+     */ 
+    function authorizeOperator(address _operator) external {
+        approvals[msg.sender][_operator] = true;
+        emit AuthorizedOperator(_operator, msg.sender);
+    }
+
+    /**
+     * @notice Revokes authorisation of an operator previously given for all partitions of `msg.sender`.
+     * NB - Allowing investors to authorize an investor to be an operator of all partitions
+     * but it doesn't mean we operator is allowed to transfer the LOCKED partition values.
+     * Logic for this restriction is written in `operatorTransferByPartition()` function.
+     * @param _operator An address which is being de-authorised
+     */
+    function revokeOperator(address _operator) external {
+        approvals[msg.sender][_operator] = false;
+        emit RevokedOperator(_operator, msg.sender);
+    }
+
+    /**
+     * @notice Authorises an operator for a given partition of `msg.sender`
+     * @param _partition The partition to which the operator is authorised
+     * @param _operator An address which is being authorised
+     */
+    function authorizeOperatorByPartition(bytes32 _partition, address _operator) external {
+        _isValidPartition(_partition);
+        partitionApprovals[msg.sender][_partition][_operator] = true;
```

Comment:
If we are only allowing people to authorize others on the UNLOCKED partition, do we need to distinguish this at all from `authorizeOperator`?

Reply:
As per the ERC definition, it should be distinguished from the `authorizeOperatorByPartition()`. I am trying to follow the same here. An investor can authorize the operator to trade every partition by calling `authorizeOperator()` but if I try to call `operatorRedeemByPartition` & `operatorTransferByPartition` with different partition beside UNLOCKED it reverts whether I am authorized to trade that partition. 
On `authorizeOperatorByPartition()` I explicitly add the valid partition check to not waste the gas of the investor by allowing the investor to authorize the operator to trade LOCKED or other partition besides UNLOCKED. By this way, I think this function also follows the standard definition as per the ERC.